src/data_access_layer/mappers/event_system/event_registration_mapper.ts

```diff
@@ -0,0 +1,167 @@
+import Result from "../../../common_classes/result"
+import EventRegistration from "../../../event_system/event_registration";
+import Mapper from "../mapper";
+import {PoolClient, QueryConfig} from "pg";
+import uuid from "uuid";
+
+const format = require('pg-format')
+
+/*
+    EventRegistrationMapper extends the Postgres database Mapper class and allows
+    the user to map a data structure to and from the attached database. The mappers
+    are designed to be as simple as possible and should not contain things like
+    validation or transformation of the data prior to storage - those operations
+    should live in a Repository or on the data structure's class itself. Also
+    try to avoid listing functions, as those are generally covered by the Repository
+    class/interface as well.
+*/
+export default class EventRegistrationMapper extends Mapper{
+    public static tableName = "registered_events";
+
+    private static instance: EventRegistrationMapper;
+
+    public static get Instance(): EventRegistrationMapper {
+        if(!EventRegistrationMapper.instance) {
+            EventRegistrationMapper.instance = new EventRegistrationMapper()
+        }
+
+        return EventRegistrationMapper.instance
+    }
+
+    public async Create(userID:string, input: EventRegistration, transaction?: PoolClient): Promise<Result<EventRegistration>> {
+        const r = await super.run(this.createStatement(userID, input), {transaction, resultClass: EventRegistration})
+        if(r.isError) return Promise.resolve(Result.Pass(r))
+
+        return Promise.resolve(Result.Success(r.value[0]))
+    }
+
+    public async Update(userID:string, input: EventRegistration, transaction?: PoolClient): Promise<Result<EventRegistration>> {
+        const r = await super.run(this.fullUpdateStatement(userID, input), {transaction, resultClass: EventRegistration})
+        if(r.isError) return Promise.resolve(Result.Pass(r))
+
+        return Promise.resolve(Result.Success(r.value[0]))
+    }
+
+    public async Retrieve(id: string): Promise<Result<EventRegistration>> {
+       return super.retrieve(this.retrieveStatement(id), {resultClass: EventRegistration})
+    }
+
+    public async List(): Promise<Result<EventRegistration[]>> {
```

Comment:
Same comment about List(). How to handle the ListBy* functionality?

Reply:
This I can move to the event registration repository.